daemon/api_snap_conf.go

```diff
@@ -91,6 +102,72 @@ func getSnapConf(c *Command, r *http.Request, user *auth.UserState) Response {
 	return SyncResponse(currentConfValues)
 }
 
+// cleanExperimentalFlags returns an exact copy of val after hiding experimental
+// features that are no longer supported from core.experimental configuration
+// generic queries (i.e. key is "" or "experimental").
+// For exact queries (e.g. core.experimental.old-flag) a client.ErrorKindConfigNoSuchOption
+// error is returned.
+//
+// This helper should only be called for core configurations. Any errors when parsing
+// core config are ignored and val is returned without modification.
+func cleanExperimentalFlags(key string, val interface{}) (interface{}, *apiError) {
```

Comment:
sorry to nitpick but I'd replace "clean" with "prune" which is a bit clearer in this context IMO